### PR TITLE
Add better cache header logic

### PR DESCRIPTION
**Pull request by nkansal96**

### CODE REVIEW

File: blaze/mahimahi/server/filestore.py [hunk 1]
```diff
@@ -68,21 +99,20 @@ def read(path: str) -> "File":
...
-        }
+        res_headers = {h.key.decode().lower(): h.value.decode() for h in record.response.header}
 
-        body = record.response.body
+        # Pushable objects must be cacheable
+        is_cacheable = check_cacheability(res_headers)
 
         # Unchunk the body if it is chunked since HTTP/2 does not support chunked encoding
+        body = record.response.body
         if TRANSFER_ENCODING_HEADER in res_headers and "chunked" in res_headers[TRANSFER_ENCODING_HEADER].lower():
```
Comment (muralisr) [inline blaze/mahimahi/server/filestore.py]: on the right side, if you unchunk the body, should we delete the transfer_encoding _header like in the old version?

File: blaze/mahimahi/server/filestore.py [hunk 2]
```diff
@@ -10,22 +10,50 @@
...
 
 
+def check_cacheability(headers: Dict[str, str]) -> bool:
+    cache_control = headers.get(CACHE_CONTROL_HEADER, "")
+    if "no-cache" in cache_control or "no-store" in cache_control:
+        return False
+
+    if "max-age=" in cache_control:
+        try:
+            max_age = int(cache_control.split("max-age=")[1].split(",")[0])
```
Comment (muralisr) [inline blaze/mahimahi/server/filestore.py]: what if cache-control header is not set but expires  or last-modified is non-zero/in the future. 
https://github.com/muralisr/mahimahi/blob/master/src/frontend/is_cacheable.cc (lines 124 to 164)

File: blaze/mahimahi/server/filestore.py (hunk 1 above)
Comment (nkansal96) [reply blaze/mahimahi/server/filestore.py]: The header has been added to `REMOVE_HEADERS` and will be removed on line 113

File: blaze/mahimahi/server/filestore.py (hunk 2 above)
Comment (nkansal96) [reply blaze/mahimahi/server/filestore.py]: If there is a `max-age=0`, we say it's not cacheable according to line 118 of the `is_cacheable.cc`. If there's no value or its set to an invalid value, then it passes and checks expires and last-modified in the last line if the function. If the `max-age` is greater than zero, we say its cacheable (line 159 of `is_cacheable.cc`)

**Pull request merged**